### PR TITLE
minor cleanup of environment-modules documentation

### PR DESCRIPTION
**Pull request by davydden**
@citibeth have a look. These changes reflect my experience on Ubuntu 14.04.

### CODE REVIEW

**Comment by citibeth** [comment]
Can you please explain the rationale behind this change?  Is there some
case where the original command I gave doesn't work?  I'm not comfortable
with giving people commands and then telling them that they might not work
and things must be hand-edited.

The rationale of the original text was to build a set of commands that are
as foolproof as possible, that users can cut-n-paste.  Everything that must
be manually checked slows down bootstrapping of Spack.

BTW... I think this kind of stuff needs a reorganization into a general
bootstrapping section (that would be for a different PR).  Spack is
super-easy to install (just git clone and run)... as long as you have
up-to-date compilers, curl, OpenSSL, Environment Modules, binutils, etc.
Otherwise, you need to install a bunch of stuff with Spack before you can
work reliably.  The bootstrapping documentation would talk about the
general kinds of things that can go wrong, and show how Spack can be used
to fix those problems.

-- Elizabeth

**Comment by davydden** [comment]
> Can you please explain the rationale behind this change?  Is there some
case where the original command I gave doesn't work?  I'm not comfortable
with giving people commands and then telling them that they might not work
and things must be hand-edited

Yes, that is exactly  my experience on Ubuntu 14.04 when I try to follow the current documentation.
Note that `${MODULES_HOME}/Modules/${MODULES_VERSION}/bin/add.modules` also was wrong at least for the current version of `environment-modules` package. Namely, the correct path is `${MODULES_HOME}/Modules/${MODULES_VERSION}/bin/add.modules`. 

> The rationale of the original text was to build a set of commands that are
as foolproof as possible, that users can cut-n-paste.  Everything that must
be manually checked slows down bootstrapping of Spack.

I agree with that. But if something apparently is not that bullet proof, i think it is better to give some hints to the user how to fix it. Otherwise `module avail` just does not work because it was trying to execute `/bin/modulecmd` instead of `full-path-to/bin/modulecmd`.

@citibeth  if you have Ubuntu 14.04 around, could you please try following the current documentation is see if you hit any of the issues I am trying to fix in this PR?

**Comment by davydden** [comment]
of course, the issues with `/Modules/bin/add.modules` script is not related to Spack, but since we tell users how to set up modules, i think it is better to discuss possible pitfalls.

**Comment by citibeth** [comment]
So... you're setting up modules on Ubuntu 14.04 using the Ubuntu-provided
version of environment-modules?  No wonder the required instructions are
different.  I suggest the following reorganization:

1. Create a "Bootstrapping" section of the manual, going over all the
things one commonly might install with Spack due to old/missing OS versions
on old OS's.  That was the intent of the docs I originally wrote.

2. Give more general instructions on setting up environment modules, no
matter where you got them form.

**Comment by davydden** [comment]
> So... you're setting up modules on Ubuntu 14.04 using the Ubuntu-provided
version of environment-modules?

ah.. No! I use spack, ubuntu has no environment modules at all.

**Comment by davydden** [comment]
to reiterate, the instructions as of now do not apply to `environment-modules` package provided by spack.

**Comment by citibeth** [comment]
Ubuntu 14.04LTS has environment modules:

http://packages.ubuntu.com/search?suite=default&section=all&arch=any&keywords=environment-modules&searchon=names

to reiterate, the instructions as of now do not apply to environment-modules
> package provided by spack.
>

Hmmm... they worked for me, installing Spack's environment-modules on
Ubuntu 12.04LTS.  This clearly needs another look.

**Comment by davydden** [comment]
> Ubuntu 14.04LTS has environment modules:
>
> http://packages.ubuntu.com/search?suite=default&section=all&arch=any&keywords=environment-modules&searchon=names

I did not know that. I checked and I don't have them installed.

> Hmmm... they worked for me, installing Spack's environment-modules on
Ubuntu 12.04LTS.  This clearly needs another look.

that's very strange... For me both on Ubuntu 14.04 and on OSX the directory structure of Spack's `environment-modules` corresponds to the changes proposed in this PR.

Could that be that you used a different version of `environment-modules` package which happened to have a different directory structure? Can you try installing the current Spack's version and see what you get?

**Comment by citibeth** [comment]
The environment-modules package was written by two people simultaneously (@citibeth and @gartung).  I later combined them into one package offering the best of both worlds.  It's likely that the instructions I gave work for environment-modules as I originally wrote it, but not for the final version with both feature sets merged in.

I unfortunately do not have the time to do those tests right now.  But I'm adding it here as a TODO list.

- [ ] Check the directory structure installed by various versions of the Spack environment-modules package, and provide instructions that match it.

**Comment by gartung** [comment]
I setup environment-modules on OSX using
source `spack location -i environment-modules`/Modules/init/sh

**Comment by davydden** [comment]
> I setup environment-modules on OSX using
> 
> source `spack location -i environment-modules`/Modules/init/sh

that confirms that current instructions are indeed wrong.

**Comment by davydden** [comment]
i will update the PR with instructions to manually add `source .../init/sh` (or equivalent) to bash as this works for me OOB.

**Comment by davydden** [comment]
@gartung I updated the PR. please, let me know if you would reword something.

**Comment by gartung** [comment]
The wording is fine with me. The formatting is not correct as it runs the two lines together.

**Comment by davydden** [comment]
@gartung thanks for pointing this out. I fixed it.

**Comment by citibeth** [comment]
I go along with this PR.  It seems to correctly document the current behavior.  If I find an added wrinkle later, I can submit that as a PR.
